### DISCUSSION TITLE
MA service calls and entities missing in HA

### DISCUSSION THREAD

**Discussion by thatso** [music-assistant/support#2757] [Q&A] [answered]
I'm currently running HAOS Core 2024.8.1, Supervisor 2024.08.0 and OS 12.4. I've installed Music Assistant Server 2.1.4 as HA add-on (not via HACS). It basically runs and shows up in the sidebar. I can play music files and radio stations.
However, the installation [docs](https://music-assistant.io/integration/installation/) say there should be service calls and MA media player entities to control MA from HA. All of them are missing. I don't see any MA media player entities in HA and there is no MA player service.
I've reinstalled, rebooted and cleared my browser cache numerous times to no avail. Now I wonder if I got something wrong or the documentation is faulty.
Searching the bug tracker and countless forum posts also did not help.
I tried to manually add MA in HA devices and integrations additionally to the add-on, but there is no such integration to choose.
In MA, the plugin provider setting for HA shows no relevant configuration options:

![image](https://github.com/user-attachments/assets/57550e94-c3c2-43ce-8165-421965fc8209)

**Comment by OzGav** [comment]
https://music-assistant.io/integration/

**Reply by thatso** [reply]
What gives? Throwing a link without explanation is not an answer. Did you even read my question? 
I've seen and read that page already. Guess what: not helping.
If you refer to "The Home Assistant Plug-in provider. This plugin is currently an empty placeholder" then what about [The integration adds three service calls for use in scripts and automations](https://music-assistant.io/integration/installation/#installation-of-the-home-assistant-integration)?

**Reply by OzGav** [reply]
So you have installed the Integration via HACS then....

**Reply by thatso** [reply]
In fact, I tried that too and MA did not even start. But that is not of interest as you obviously enjoy trolling.
Let me summarize: the HA blog boldly wrote in [May 2024](https://www.home-assistant.io/blog/2024/05/09/music-assistant-2/): "After many years of development and testing with the amazing community, we can finally say this is ready for daily use" and "Music Assistant is distributed as a Home Assistant add-on". Three months later it turns out that the documentation is loaded with contradictions starting with the proper way to even install MA. Great work, guys! I've already wasted enough of my time trying to get this crap to run before I opened this request for help and now a so-called maintainer won't give a useful answer. :angry:

**Reply by OzGav** [reply]
You need to calm down. In your initial post you made no mention of the integration, in fact it seemed clear you misunderstand the difference between the ADDON and the integration which is why I linked to the description of what the integration is. 

So I think what you are not understanding, which is explained on the home page https://music-assistant.io/#architecture, is that you need BOTH the addon and the integration

**Reply by thatso** [reply]
Are you saying after installing the add-on in HA the integration has to be added via HACS additionally? That is totally unexpected and should be emphasized in the docs.
I assumed that either an installation via HA itself or HACS would provide both components as usual.

**Reply by OzGav** [reply, marked as answer]
Correct. The home page says:

>Music Assistant consists of multiple building blocks:

>Music Assistant Server ([Installation Instructions](https://music-assistant.io/installation/))
Music Assistant Integration for Home Assistant ([Installation Instructions](https://music-assistant.io/integration/installation/))

You will notice we have placed the link to both sets of installation instructions. 

Just after that there is further information about what these two things are:

> Music Assistant Server
The Music Assistant server is a free, opensource Media library manager that connects to your streaming services and a wide range of connected speakers. 

> Music Assistant Integration
Connects Home Assistant to your Music Assistant Server to allow control from your HA instance, allow you to automate your music and allows voice control! The Integration also allows the exposure of HA media players to MA furthering the options you have for playback.

Then on the server installation page it says

> Music Assistant (MA) can be operated as a complete standalone product but it is actually designed to use side by side with Home Assistant. MA is built with automation in mind, hence the recommended installation method is to run the server as a Home assistant Add-on and then optionally [add the HA integration](https://music-assistant.io/integration/installation/).

Then there is a whole page describing what the Integration does. So I am not sure what else would make this clearer.

**Reply by thatso** [reply]
As said: for any other component HA and HACS are mutually exclusive so naturally I expected any single one of them to provide all needed parts. Specifically after the announcement "Music Assistant is distributed as a Home Assistant add-on" it seemed logical that before that, MA was only available via HACS and now it is an official add-on.

**Reply by OzGav** [reply]
Always good not to assume.

**Reply by thatso** [reply]
OK, closing this as solved. But I still think this two-part installation is unexpected.
